src/pages/index.js

```diff
@@ -1,111 +1,227 @@
-console.log("Script is loaded"); // Add this at the top of your JavaScript file
-
-
-import './index.css'
-import UserInfo from "../components/UserInfo.js"
+import './index.css';
+import UserInfo from "../components/UserInfo.js";
 import Card from "../components/Card.js";
 import FormValidator from "../components/FormValidator.js";
-import { initialCards } from "./utils/constants.js";
 import Section from "../components/Section.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import PopupWithImage from "../components/PopupWithImage.js";
-import {editProfileButton, addCardButton, addCardFormSelector, previewModal, editProfileSelector, cardSelector, config, editProfileForm, addCardModalSelector, addCardFormElement} from "./utils/constants.js"
-
-
-
+import Api from '../components/Api.js';
+import { editProfileButton, addCardButton, previewModal, editProfileSelector, cardSelector, config, editProfileForm, addCardModalSelector, addCardFormElement } from "./utils/constants.js";
+
+// Initialize API
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "3058e262-bd72-484f-aef9-41d0e62ea999",
+    "Content-Type": "application/json",
+  },
+});
+
+// Store current user ID
+let currentUserId;
+
+// Initialize user info
 const userInfo = new UserInfo({
   nameSelector: "#profileName",
-  hobbySelector: "#profileDescription"
-})
-
-const cardSection = new Section({
-  items: initialCards,
-  renderer: (item) => {
-    const cardElement =  createCard(item);
-    cardSection.addElement(cardElement);
-  }
-},
-".cards__list"
+  hobbySelector: "#profileDescription",
+});
+
+// Initialize section for cards
+const cardSection = new Section(
+  {
+    items: [],
+    renderer: (item) => {
+      const cardElement = createCard(item);
+      cardSection.addElement(cardElement);
+    },
+  },
+  ".cards__list"
 );
 
+// Initialize Popups
+const popupWithImage = new PopupWithImage(previewModal);
+popupWithImage.setEventListeners();
 
-
-const popupWithForm = new PopupWithForm(editProfileSelector, (formData) => {
-  userInfo.setUserInfo({
-    name: formData.title,
-    job: formData.description
-  });
-})
-popupWithForm.setEventListeners();
+const editProfileModal = new PopupWithForm(editProfileSelector, (inputValues) => {
+  editProfileModal.setLoading(true, "Saving...");
```

Comment:
**Could be improved**
If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}

```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```